### PR TITLE
Merge remote-tracking branch 'cocos-for-editor/v3.8.1' into develop

### PR DESCRIPTION
**Pull request by knoxHuang**
Re: #

### Changelog

* Merge remote-tracking branch 'cocos-for-editor/v3.8.1' into develop

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/external-config.json [hunk 1]
```diff
@@ -3,6 +3,6 @@
         "type": "github",
         "owner": "cocos-creator",
         "name": "engine-native-external",
-        "checkout": "develop-25"
+        "checkout": "v3.8.1-4"
```
Comment (knoxHuang) [inline native/external-config.json]: 这块是冲突的修改，麻烦看一下是否处理正确

File: native/cocos/renderer/pipeline/custom/NativeRenderQueue.cpp
Comment (knoxHuang) [inline native/cocos/renderer/pipeline/custom/NativeRenderQueue.cpp:1]: 这个是冲突的脚本，看看是否修改正确

File: native/external-config.json (hunk 1 above)
Comment (minggo) [reply native/external-config.json]: 第三方库需要从 3.8.1 同步到 develop 分支，然后再修改这个值。

File: native/cocos/renderer/pipeline/custom/NativeRenderQueue.cpp
Comment (minggo) [reply native/cocos/renderer/pipeline/custom/NativeRenderQueue.cpp:1]: @star-e

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
@star-e i merged first as this PR is easy to conflict. Please send a PR for [native/cocos/renderer/pipeline/custom/NativeRenderQueue.cpp](https://github.com/cocos/cocos-engine/pull/15703/files/5d75efd558e12618d043419323d3e76cfe57031d#diff-fc545b61ca348dc794565108066c87a254a8e627cd304a2366ba44794aa48c54) if needed.

File: native/cocos/renderer/pipeline/custom/NativeRenderQueue.cpp
Comment (star-e) [reply native/cocos/renderer/pipeline/custom/NativeRenderQueue.cpp:1]: it looks fine. @minggo

**Pull request merged**